MAID-2590: fix netsim tests

povilasb opened the pull request:
Makes multiple changes that fix `tests::netsim::tcp_bootstrap_over_poor_connection`.

povilasb committed:
refactor/tests: reduce duplicate code

povilasb committed:
fix/palistener: increase initial message timeout

Netsim based tests with packet loss showed that 3 seconds might not be enough to do connection handshake. Some testing shows that 5 seconds is good enough, but I see no reason why we can't wait for 10 seconds before timing out.

povilasb committed:
refactor/tests: use more appropriate naming

povilasb committed:
fix/tests: use unique bootstrap cache per test instance

This test used to use default bootstrap cache which resulted in shared cache and eventually failing tests.

povilasb committed:
fix/tests: use production heartbeat timeouts for netsim tests

Too low heartbeat timeouts make netsim tests with packet loss fail. We aim to test as close to production as possible with netsim, hence production like heatbeat timeouts make more sense in this case.

maidsafe-highfive commented:
Thanks for the pull request, and welcome! The MaidSafe team is excited to review your changes, and you should hear from @canndrew (or someone else) soon.

If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code. The way GitHub handles out-of-date commits, this should also make it reasonably obvious what issues have or haven't been addressed. Large or tricky changes may require several passes of review and changes.

Please see [CONTRIBUTOR.md](../blob/master/CONTRIBUTOR) for more information.

povilasb committed:
feat/tests: split test cases into two each: uTP and TCP

Shows better what's the reason in case of failure.

povilasb committed:
chore/appveyor: disable netsim feature

Some code/constants depend on netsim feature enabled. Make sure this feature is off when running tests on Windows. Otherwise some inactivity tests fail because of different inactivity timeouts for netsim tests.
